source/dub/recipe/sdl.d

```diff
@@ -343,64 +342,79 @@ private string expandPackageName(string name, string parent_name, Tag tag)
 
 private string stringTagValue(Tag t, bool allow_child_tags = false)
 {
-	import std.string : format;
 	enforceSDL(t.values.length > 0, format("Missing string value for '%s'.", t.fullName), t);
 	enforceSDL(t.values.length == 1, format("Expected only one value for '%s'.", t.fullName), t);
-	enforceSDL(t.values[0].peek!string !is null, format("Expected value of type string for '%s'.", t.fullName), t);
 	enforceSDL(allow_child_tags || t.tags.length == 0, format("No child tags allowed for '%s'.", t.fullName), t);
 	// Q: should attributes be disallowed, or just ignored for forward compatibility reasons?
 	//enforceSDL(t.attributes.length == 0, format("No attributes allowed for '%s'.", t.fullName), t);
-	return t.values[0].get!string;
+	return t.values[0].expect!string(t);
+}
+
+private T expect(T)(
+	Value value,
+	Tag errorInfo,
+	string customFieldName = null,
+	string file = __FILE__,
+	int line = __LINE__
+)
+{
+	return value.match!(
+		(T v) => v,
+		(fallback)
+		{
+			enforceSDL(false, format("Expected value of type " ~ T.stringof ~ " for '%s', but got %s.",
+				customFieldName.length ? customFieldName : errorInfo.fullName,
+				typeof(fallback).stringof),
+				errorInfo, file, line);
+			return T.init;
+		}
+	);
 }
 
 private string[] stringArrayTagValue(Tag t, bool allow_child_tags = false)
 {
-	import std.string : format;
 	enforceSDL(allow_child_tags || t.tags.length == 0, format("No child tags allowed for '%s'.", t.fullName), t);
 	// Q: should attributes be disallowed, or just ignored for forward compatibility reasons?
 	//enforceSDL(t.attributes.length == 0, format("No attributes allowed for '%s'.", t.fullName), t);
 
 	string[] ret;
-	foreach (v; t.values) {
-		enforceSDL(t.values[0].peek!string !is null, format("Values for '%s' must be strings.", t.fullName), t);
-		ret ~= v.get!string;
+	foreach (i, v; t.values) {
+		ret ~= v.expect!string(t, text(t.fullName, "[", i, "]"));
 	}
 	return ret;
 }
 
-private void parsePlatformStringArray(Tag t, ref string[][string] dst)
+private string getPlatformSuffix(Tag t, string file = __FILE__, int line = __LINE__)
 {
 	string platform;
 	if ("platform" in t.attributes)
-		platform = t.attributes["platform"][0].value.get!string;
-	dst[platform] ~= t.values.map!(v => v.get!string).array;
+		platform = t.attributes["platform"][0].value.expect!string(t, t.fullName ~ " platform", file, line);
+	return platform;
```

Comment:
In the future I'd recommend writing this as:
```D
if (auto ptr = "platform" in t.attributes)
    return (*ptr)[0].value...
return null;
```
To make it more obvious what is happening, but you are just matching the old style here.